File: js/13164.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const [N, K] = inputs[0].split(' ').map(Number);
const arr = inputs[1].split(' ').map(Number);

const diff = [];

for (let i = 0; i < N - 1; i++) {
  diff.push(arr[i + 1] - arr[i]);
}

diff.sort((a, b) => a - b);

let ans = 0;
for (let i = 0; i < N - K; i++) {
  ans += diff[i];
}

console.log(ans);

File: c++/13164.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  int N, K;
  cin >> N >> K;

  vector<int> v;

  for (int i = 0; i < N; i++) {
    int a;
    cin >> a;
    v.push_back(a);
  }

  vector<int> diff;

  for (int i = 0; i < N - 1; i++) {
    diff.push_back(v[i + 1] - v[i]);
  }

  sort(diff.begin(), diff.end());

  int ans = 0;

  for (int i = 0; i < N - K; i++) {
    ans += diff[i];
  }

  cout << ans;

  return 0;
}